src/i_oalsound.c

```diff
@@ -0,0 +1,660 @@
+    }
+
+#ifdef ALC_OUTPUT_MODE_SOFT
+    if (alcIsExtensionPresent(oal->device, "ALC_SOFT_output_mode") == AL_TRUE)
+    {
+        attribs[i++] = ALC_OUTPUT_MODE_SOFT;
+        attribs[i++] = use_3d ? (snd_hrtf ? ALC_STEREO_HRTF_SOFT : ALC_ANY_SOFT) :
+                                ALC_STEREO_BASIC_SOFT;
+    }
+#endif
+}
+
+boolean I_OAL_InitSound(boolean use_3d)
+{
+    ALCdevice *device;
+    ALCint attribs[OAL_NUM_ATTRIBS];
+
+    if (oal)
+    {
+        I_OAL_ShutdownSound();
+    }
+
+    if (!OpenDevice(&device))
+    {
+        fprintf(stderr, "I_OAL_InitSound: Failed to open device.\n");
+        return false;
+    }
+
+    oal = calloc(1, sizeof(*oal));
+    oal->device = device;
+    GetAttribs(use_3d, attribs);
+
+    oal->context = alcCreateContext(oal->device, attribs);
+    if (!oal->context || !alcMakeContextCurrent(oal->context))
+    {
+        fprintf(stderr, "I_OAL_InitSound: Error creating context.\n");
+        I_OAL_ShutdownSound();
+        return false;
+    }
+
+    oal->sources = malloc(sizeof(*oal->sources) * MAX_CHANNELS);
+    alGetError();
+    alGenSources(MAX_CHANNELS, oal->sources);
+    if (!oal->sources || alGetError() != AL_NO_ERROR)
+    {
+        fprintf(stderr, "I_OAL_InitSound: Error creating sources.\n");
+        I_OAL_ShutdownSound();
+        return false;
+    }
+
+    oal->SOFT_source_spatialize = alIsExtensionPresent("AL_SOFT_source_spatialize") == AL_TRUE;
+    oal->EXT_EFX = alcIsExtensionPresent(oal->device, "ALC_EXT_EFX") == AL_TRUE;
+    oal->EXT_SOURCE_RADIUS = alIsExtensionPresent("AL_EXT_SOURCE_RADIUS") == AL_TRUE;
+    oal->active = malloc(sizeof(*oal->active) * MAX_CHANNELS);
+    InitDeferred();
+    ResetParams();
+
+    return true;
+}
+
+boolean I_OAL_ReinitSound(boolean use_3d)
+{
+    LPALCRESETDEVICESOFT alcResetDeviceSOFT = NULL;
+    ALCint attribs[OAL_NUM_ATTRIBS];
+
+    if (!oal)
+    {
+        fprintf(stderr, "I_OAL_ReinitSound: OpenAL not initialized.\n");
+        return false;
+    }
+
+    if (alcIsExtensionPresent(oal->device, "ALC_SOFT_HRTF") != AL_TRUE)
+    {
+        fprintf(stderr, "I_OAL_ReinitSound: Extension not present.\n");
+        return false;
+    }
+
+    alcResetDeviceSOFT = FUNCTION_CAST(LPALCRESETDEVICESOFT,
+                                       alGetProcAddress("alcResetDeviceSOFT"));
+
+    if (!alcResetDeviceSOFT)
+    {
+        fprintf(stderr, "I_OAL_ReinitSound: Function address not found.\n");
+        return false;
+    }
+
+    GetAttribs(use_3d, attribs);
+
+    if (alcResetDeviceSOFT(oal->device, attribs) != AL_TRUE)
+    {
+        fprintf(stderr, "I_OAL_ReinitSound: Error resetting device.\n");
+        I_OAL_ShutdownSound();
+        return false;
+    }
+
+    ResetParams();
+
+    return true;
+}
+
+boolean I_OAL_AllowReinitSound(void)
+{
+    if (!oal)
+    {
+        return false;
+    }
+
+    return alcIsExtensionPresent(oal->device, "ALC_SOFT_HRTF") == AL_TRUE;
```

Comment:
Why does re-init depend on the presence of the "ALC_SOFT_HRTF" extension? Also, parentheses. 😉

Reply:
`alcResetDeviceSOFT()` was added as part of the `ALC_SOFT_HRTF` extension. Added a comment here https://github.com/fabiangreffrath/woof/pull/1118/commits/9d885fc0919a729b021f697dc2ca451bc23efd22

References:
https://openal-soft.org/openal-extensions/SOFT_HRTF.txt
https://github.com/kcat/openal-soft/blob/master/include/AL/alext.h#L436-L458

Added missing parentheses here: https://github.com/fabiangreffrath/woof/pull/1118/commits/516decc26dafba72f9eb2efa4c6ae027f68f3356

Reply:
Awesome, thanks!